sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -421,6 +448,21 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
     }
     newOrdering(order, Seq.empty)
   }
+
+  /**
+   * If a plan contains a RecursiveReferenceExec without an enclosing RecursiveTableExec than it
```

Comment:
it is basically a copy of the other one, we need a better solution here.

Reply:
Changed a bit to use `lazy val` but that doesn't really address your comment. Shall I extract a common trait and use it in `QueryPlan` to avoid code duplication?

Reply:
Can't we just modify the QueryPlan's `sameResult` method and keep it final as it was?

Reply:
ok, I will change it a bit

Reply:
done